configure

```diff
@@ -0,0 +1,360 @@
+        else
+            sub_info No.
+        fi
+    fi
+}
+
+try_ld_argument() {
+    info Checking whatever linker accepts $2...
+    if $compiler $CPPFLAGS $CXXOPTS $CUSTOM_CXXOPTS $2 testfile.cc -o testfile.o > /dev/null 2>&1 ; then
+        sub_info Yes.
+        eval "$1=\"\${$1} \$2\""
+        eval "$1=\${$1# }"
+    else
+        sub_info No.
+    fi
+}
+
+usage() {
+    cat << EOF
+Usage: $0 [OPTION]...
+
+Defaults for the options are specified in brackets.
+
+  --help                        This help message.
+  --quiet                       Don't print normal messages, just errors.
+  --clean                       Clear mconfig and configure's temp files.
+
+Target options:
+  --target                      Set the target system (for cross-compiling) []
+  --host                        Same as --target
+  --platform                    Set the platform manually (Just for cross-platform cross-compile!) [autodetected]
+                                  For all cross-compiles (even this) don't forget to set correct --target!
+
+Installation directories:
+  --prefix=PREFIX               Main installtion prefix [/usr]
+  --exec-prefix=EPREFIX         Main executables location [/]
+  --sbindir=SBINDIR             Dinit executables [EPREFIX/sbin]
+  --mandir=MANDIR               Dinit man-pages location [PREFIX/share/man]
+  --syscontrolsocket=SOCKETPATH Dinitctl socket location [/run/dinitctl]
+
+Optional options:
+  --shutdown-prefix=PREFIX      Name prefix for shutdown, poweroff, reboot, halt programs []
+  --enable-shutdown             Build shutdown, poweroff, reboot, halt programs [Enabled on Linux based systems]
+  --disable-shutdown            Don't build shutdown, poweroff, reboot, halt programs
+  --enable-cgroups              Enable Cgroups support [Enabled only on Linux based systems]
+  --disable-cgroups             Disable Cgroups support
+  --enable-sanitizers           Enable -fsanitize in Unit tests [Enabled on supported systems]
+  --disable-sanitizers          Disable -fsanitize in Unit tests
+  --enable-utmpx                Enable manipulating the utmp/utmpx database via the related POSIX functions [auto]
+  --disable-utmpx               Disable manipulating the utmp/utmpx database via the related POSIX functions
+  --disable-cxxopts             (Not recommanded) Disable ALL CXXOPTS even -std=c++11
+                                  Don't forget set your flags via CXXOPTS & CXXFLAGS envs otherwise
+                                  build may be failed!
+  --disable-ldflags             (Not recommanded) Disable ALL LDFLAGS even "-lrt on FreeBSD"
+                                  You can use LDFLAGS and CUSTOM_LDFLAGS envs to set your ld flags.
+
+Environment variables can be used:
+  CXX                           If you wanna use specific C++ compiler.
+  CPPFLAGS                      If you wanna use some arguments in C++ compiler command line.
+                                  It's passed before CXXOPTS
```

Comment:
I'll fix this thing.
> Also, CPPFLAGS are traditionally preprocessor flags. If this is just intended as a way to supply extra flags then it should probably be named something else. But also, is it necessary at all?

Also, I think yes, because sometimes you need to set a compiler flags before `CXXOPTS` (For example seems like Debian for "Hardening" do this thing).

Reply:
Can you give me an example of an option that needs to go before the other options? Typically later options override earlier ones so there's no advantage to having options go earlier on the command line

Reply:
Seems like you right.